jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -128,7 +128,7 @@ def link_data_in_table(app, data_obj=None):
             wc_spec_1d = app.session.data_collection[spec_1d].world_component_ids
             wc_spec_2d = app.session.data_collection[spec_2d].world_component_ids
 
-            wc_spec_ids.append(LinkSame(wc_spec_1d[0], wc_spec_2d[1]))
+            wc_spec_ids.append(LinkSame(wc_spec_1d[0], wc_spec_2d[-1]))
```

Comment:
This fixes Kyle's original problem. The extracted spectra did not have "Offset" in world component IDs, so the index changed. But this fix isn't enough.